src/realm/object-store/shared_realm.cpp

```diff
@@ -1281,6 +1288,20 @@ void Realm::close()
     m_config = {};
 }
 
+void Realm::terminate_sync_and_delete_realm_files()
+{
+    try {
```

Comment:
I have changed the logic a little: 
1. shutdown sync session if needed 
2. copy path
3. close realm
4. Delete Realm files